### PR TITLE
Fix /api/repo/search integration tests

### PR DESCRIPTION
**Pull request by Morlinest**
Extracted fix from #2371. Fixes (and improves) #2446.

### CODE REVIEW

File: integrations/api_repo_test.go [hunk 1]
```diff
@@ -33,22 +32,19 @@ func TestAPIUserReposNotLogin(t *testing.T) {
...
 
 	req := NewRequestf(t, "GET", "/api/v1/repos/search?q=%s", keyword)
 	resp := MakeRequest(t, req, http.StatusOK)
 
-	var body searchResponseBody
+	var body api.SearchResults
 	DecodeJSON(t, resp, &body)
-	for _, repo := range body.data {
-		assert.True(t, strings.Contains(repo.Name, keyword))
+	assert.NotEmpty(t, body.Data)
```
Comment (bkcsoft) [inline integrations/api_repo_test.go:44]: `require.NotEmpty` ?

Comment (Morlinest) [reply integrations/api_repo_test.go:44]: @bkcsoft I don't want to stop tests.

Comment (daviian) [reply integrations/api_repo_test.go:44]: @Morlinest Why? One failing test is enough to let whole integration tests fail. It will speed up testing a lot.

Comment (Morlinest) [reply integrations/api_repo_test.go:44]: @daviian If you are thinking about that like this, we should use `require.*` everywhere :). I want to see more than 1 step ahead. In my opinion, it's like parsing a form and showing just first error to user.

Comment (daviian) [reply integrations/api_repo_test.go:44]: @Morlinest in this particular test every following assertion will evaluate to false so there is no reason to check them anyway

**Comment by ethantkoenig** [comment]
LGTM

File: integrations/api_repo_test.go (hunk 1 above)
Comment (Morlinest) [reply integrations/api_repo_test.go:44]: @daviian Following assertions are inside range loop.

**Comment by daviian** [comment]
LGTM

**Comment by lafriks** [comment]
Make LG-TM work

**Comment by lunny** [comment]
Make L-G-T-M work again

**Comment by lafriks** [comment]
Make L-GTM work :)

**Pull request merged**